### ISSUE TITLE
Cannot build TomoPy for conda-forge because of submodule

### ISSUE THREAD

**Issue by carterbox** [tomopy/tomopy#410]
`conda-forge` builds [must be created from release tarballs](https://conda-forge.org/docs/maintainer/adding_pkgs.html), but GitHub does not include submodules in release tarballs. Therefore, the PTL submodule is unavailable when building from the release tarball. This prevents me from creating a `conda-forge` release for 1.5.x. Here are some approaches that might work:

1. Release `libptl` as a separate package of headers and shared libraries on `conda-forge`. Requires a stable PTL release.
2. I could grab a PTL release tarball from GitHub and [combine with the TomoPy release](https://conda.io/projects/conda-build/en/latest/resources/define-metadata.html#source-from-multiple-sources). Requires a stable PTL release, but no conda build.

In related news, I tried to build without PTL, but the `--disable-tasking` flag is broken? My expectation is that this flag makes PTL not required. However, there are some include statements for PTL and CMake directives which are not disabled by this flag and which cause compilation to fail. For example, if `source/PTL` is empty the following compile error occurs:
```
/opt/miniconda/conda-bld/tomopy_1558640254977/work/source/include/macros.hh:101:10: fatal error: PTL/AutoLock.hh: No such file or directory
 #include "PTL/AutoLock.hh"
          ^~~~~~~~~~~~~~~~~
```

**Comment by jrmadsen** [comment]
I don't think that's true about conda-forge and release builds. PyCTest uses submodules so my conda-forge pipeline provides the git revision id

**Comment by jrmadsen** [comment]
It is `--disable-tasking`, not `--disable-ptl`. TomoPy uses a handful of headers from PTL that have full definitions in the headers, thus don't require building PTL to use them

**Comment by carterbox** [comment]
OK. I'll use something like for this the conda-forge build. If you did it for `pyctest`, building from git is probably just discouraged, but not prohibited.
```yaml
source:
  git_url: https://github.com/ilanschnell/bsdiff4.git
  git_rev: 1.1.4
```

I *did* use `--disable-tasking`. The build fails with the compiler error [above](https://github.com/tomopy/tomopy/issues/410#issue-447849195) when you download a release because `source/PTL` is empty. None of the PTL source files are included, so you can't use the PTL headers. My suggestion is that either `--disable-tasking` completely removes the PTL dependency or the required headers are added to the TomoPy repo.

**Comment by jrmadsen** [comment]
I know it fails. My point was that I named the option `--disable-tasking` because that implies something different than `--disable-ptl`. You aren't removing the dependency on PTL, you just aren't building PTL, thus, no tasking is available. If you add the required headers to TomoPy, you are more likely to add issues because you would then have a whole bunch of functions and classes with the exact same signature which would then require careful include statements and more if/else preprocessor blocks. 

Then merge what I am about to commit to 1.5.x. Then create a "release" branch. And in that branch, add this to the top-level CMakeLists.txt:

```
set(PTL_RELEASE tomopy-v1.0)
set(PTL_SHA512  38412b950a0608a15ca2a7325134cbb1d929df7d1eb209765121db0df6ab9fc5f7908f9b78a4739aa6e01954ec35fa89f31019e371198d1f526c3b7a20c50e61)
```
NOTE: the above should ONLY exist in the "release" branch. Have everyone continue to use the master and then pull the master onto the release branch.

Finally, create a 1.5.1 release from the "release" branch.

**Comment by carterbox** [comment]
> My point was that I named the option --disable-tasking because that implies something different than --disable-ptl. 

Got it. Thanks for clearing that up for me.

Using a git_url checkout on the conda-forge feedstock worked, so we don't need a special release branch. Closes #411 

Over the weekend, I was able to build for all the target systems without CUDA. I will be working on figuring out the best way to distribute the CUDA build over the next month.
